lib/openscap_parser/profile.rb

```diff
@@ -20,8 +22,23 @@ def description
     end
 
     def selected_rule_ids
-      @selected_rule_ids ||= @parsed_xml.xpath("select[@selected='true']/@idref") &&
-        @parsed_xml.xpath("select[@selected='true']/@idref").map(&:text)
+      # Look for selected rule ids where the idref contains '_rule_' that is not preceded by 'group'
+      @selected_rule_ids ||= @parsed_xml.xpath("select[@selected='true']
+                                                      [regex(@idref, '^((?!_group_).)*?(_rule_).*$')]
+                                                      /@idref", RegexHandler.new) &&
+                             @parsed_xml.xpath("select[@selected='true']
+                                                      [regex(@idref, '^((?!_group_).)*?(_rule_).*$')]
+                                                      /@idref", RegexHandler.new).map(&:text)
+    end
+
+    def selected_group_ids
+      # Look for selected group ids where the idref contains '_group_' that is not preceded by 'rule'
+      @selected_group_ids ||= @parsed_xml.xpath("select[@selected='true']
+                                                       [regex(@idref, '^((?!_rule_).)*?(_group_).*$')]
+                                                       /@idref", RegexHandler.new) &&
```

Comment:
No need for `new` if the method is static:
```suggestion
                                                       /@idref", RegexHandler) &&
```